### PR TITLE
View Refactor

### PR DESCRIPTION
**Pull request by lbwexler**
+ Dry up the Row classes with inheritance.
+ Remove the weird inversion of the meta objects
+ Add support for omitFn

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [X] Caught up with `develop` branch as of last change.
- [X] Added CHANGELOG entry, or determined not required.
- [X] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [X] Updated doc comments / prop-types, or determined not required.
- [X] Reviewed and tested on Mobile, or determined not required.
- [X] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: data/cube/impl/BaseRow.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        this.children?.forEach(it => it.noteBucketed(bucketSpec, bucketVal));
+    }
+
+    // Determine what should be exposed as the actual children in the
+    // row data.  This where we lock, skip degenerate rows, etc.
+    applyVisibleChildren() {
+        let {children, view, data} = this,
+            {lockFn, omitFn} = view.cube;
+
+        if (!children) return null;
```
Comment (haynesjm42) [inline data/cube/impl/BaseRow.js:47]: I think this should just be a return

File: data/cube/impl/BaseRow.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+        // Remove all children from the data if the query is not configure to include leaves and
+        // this row has leaves as children
+        if (!view.query.includeLeaves && children[0]?.isLeaf) {
```
Comment (haynesjm42) [inline data/cube/impl/BaseRow.js:51]: I wonder if we have a potential bug here. Say we are bucketing leaf rows but `includeLeaves` is `false` in the query - and the first child here is a bucket row containing only leaves. In that case we would not remove any of those leaves, right? Should this be a `chilldren.some(it => it.isLeaf)` ?

File: data/cube/impl/BaseRow.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+                return;
+            }
+        }
+
+        // ...otherwise wire up to your own children's data
+        data.children = children.map(it => it.data);
+    }
+
+    isRedundantChild(row) {
+        // TODO:  put this test in application code omitFn instead?
```
Comment (haynesjm42) [inline data/cube/impl/BaseRow.js:82]: I could see this being an option on the Query - `cullRedundantChildren` or something

**Comment by lbwexler** [comment]
John and I reviewed this carefully.  Its a little late in the release, but we think its an important cleanup, and gives us the all-important omitFn

File: data/cube/impl/BaseRow.js (hunk 2 above)
Comment (lbwexler) [reply data/cube/impl/BaseRow.js]: we resolved -- we don't create buckets of leaves

File: data/cube/impl/BaseRow.js (hunk 3 above)
Comment (lbwexler) [reply data/cube/impl/BaseRow.js]: yep -- let's see how things shake out

**Pull request merged**